### DISCUSSION TITLE
Being able to search file with filename to tag them

### DISCUSSION THREAD

**Discussion by WawThisGuyIsSoCool** [TagStudioDev/TagStudio#632] [General]
Hello first thanks for the soft,
I have a question, i'd like to be able to search into my huges folder of thousands of image to be able to tag them.

In the filename i often have things that could help me to sort them and tag them in bulk.
For example, i have a lot of manga images. the filename could be "dragonball, goku, akira toriyama, scan.jpg"
i'd like to be able to search "dragonball" and everything that appear tag them with "manga" "dragonball" "shonen" "scan" etc...

Is there a way to do that ? Or would it be to add search by filename into TagStudio?

Thanks again and i hope there is a way because i can't really use it otherwise :/

**Comment by Computerdores** [comment]
This has been implemented in #606 as path search. Once 9.5 releases you will be able to search like this: `path:*goku*` for any file that has `goku` in its path (This would also include `goku/other.txt`)

(You could also try this in 9.4 I think it is supposed to be implemented, however I am not sure whether it works in that version)

**Reply by WawThisGuyIsSoCool** [reply]
Ok it seems not to work, when typing "path:*xxxxx*" or "path:xxxx" or "path:<xxxxx>" (imagine "xxxxx" is "goku", or somethin in the filename)
Maybe is there an other way ?
And what do you mean by "goku/other.txt"

**Reply by Computerdores** [reply]
> Ok it seems not to work, when typing "path:_xxxxx_" or "path:xxxx" or "path:" (imagine "xxxxx" is "goku", or somethin in the filename) Maybe is there an other way ?

Oh I just remembered, you should be able to use `filename:goku` in that version and that should work (note that there is a bug with that one which means it won't show you `goku.png` if it is in a directory that also has `goku` in its name.

> And what do you mean by "goku/other.txt"

The path to a file. In this case that `other.txt` would also be found if it is in a directory called `goku`

**Reply by WawThisGuyIsSoCool** [reply]
Ok thanks for your anwser, but look, `filename:` seems not to be working or am i doing something bad ?
Here is my directory and my tagstudio window
https://imgur.com/a/Sb70rjH

even if I rename an image simply "goku.png" (without anything else (like "dragon ball, goku.jpg")) it doesn't show anything

**Reply by Computerdores** [reply]
It's most likely just broken then, I would look into fixing it, but that code will be completely replaced in 9.5.

**Reply by WawThisGuyIsSoCool** [reply]
Thank you ! when do you plan to release ?

**Reply by Computerdores** [reply]
There isn't a planned date to my knowledge, but I think @CyanVoxel can say more about that than I can
